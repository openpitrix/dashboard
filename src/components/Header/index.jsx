import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { NavLink } from 'react-router-dom';
import { observer, inject } from 'mobx-react';
import { ButtonDropdown, DropdownToggle, DropdownItem, DropdownMenu } from 'reactstrap';
import { getLoginUser } from 'src/utils';

import Logo from '../Logo';
import Input from '../Base/Input';
import styles from './index.scss';

@inject('rootStore')
@observer
export default class Header extends Component {
  state = {
    isHome: PropTypes.bool,
    dropdownOpen: false
  };

  toggleDropdown = () => {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    });
  };

  renderLoginButton() {
    const loggeInUser = getLoginUser();

    if (!loggeInUser) {
      return <NavLink to="/login">Sign In</NavLink>;
    }

    return (
      <ButtonDropdown
        className={styles.loginButton}
        isOpen={this.state.dropdownOpen}
        toggle={this.toggleDropdown}
      >
        <DropdownToggle caret className={styles.toggleBtn}>
          {loggeInUser}
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem style={{ padding: 0 }}>
            <a href="/logout">Logout</a>
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
    );
  }

  render() {
    const {
      isHome,
      rootStore: { fixNav }
    } = this.props;
    const isDark = !isHome || fixNav;
    const logoUrl = isDark ? '/assets/logo2.svg' : '/assets/logo.svg';

    return (
      <div className={classnames('header', styles.header, { [styles.darkHeader]: !isDark })}>
        <div className={styles.wrapper}>
          <Logo className={styles.logo} url={logoUrl} />
          {isDark && (
            <Input.Search className={styles.search} placeholder="Search apps in Pitrix..." />
          )}
          <div className={styles.menus}>
            <NavLink to="/apps">Catalog</NavLink>
            <NavLink to="/deployment">Deployment</NavLink>
            <NavLink to="/manage">{getLoginUser() ? 'Manage' : 'Development'}</NavLink>
            {this.renderLoginButton()}
          </div>
        </div>
      </div>
    );
  }
}
