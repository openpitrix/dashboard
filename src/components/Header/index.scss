@import "~scss/vars";

$profile-hover-color: #efe6f8;

@keyframes deepHeader {
  from {
    opacity: 0;
    background: transparent;
  }
  to {
    opacity: 1;
    background: linear-gradient(to left, $P75, #854fb9 32%, #484999);
  }
}

.header {
  position: absolute;
  top: 0;
  width: 100%;
  height: $header-height;
  z-index: 999;
  background: none;

  &.deep {
    background: linear-gradient(to left, $P75, #854fb9 32%, #484999);
  }

  &.deep, &.deepHome {
    position: fixed;

    .login{
      color: $N0;
    }

    .user [class*='target']{
      color: $N0;

      :global{
        .icon-caret-down svg{
          --primary-color: #{$N0};
          --secondary-color: #{$N0};
        }
      }
    }
  }


  &.deepHome {
    animation-duration: 0.5s;
    animation-name: deepHeader;
    animation-iteration-count: 1;
    animation-timing-function: ease-in-out;
    animation-fill-mode: forwards;
  }

  .wrapper {
    width: $content-width;
    height: $header-height;
    margin: 0 auto;
  }

  .logo {
    float: left;
    width: auto;
    height: $logo-height;
    display: inline-block;
    margin-top: $header-height / 2;
    transform: translateY(-50%);
    vertical-align: middle;
  }

  .search {
    float: right;
    display: inline-block;
    min-width: 240px;
    margin-right: 20px;
    margin-top: $header-height / 2;
    transform: translateY(-50%);
    vertical-align: top;

    input {
      box-sizing: border-box;
      width: 240px;
      color: $P45;
      font-weight: 400;
      font-size: $size-normal;
      border-radius: 16px;
      border: 1px solid transparent;
      background-color: rgba($N0, 0.1);
      //transition: color .5s ease-in;
      transition: all .3s ease-in-out;

      &::placeholder {
        font-size: 14px;
        color: $P45;
        line-height: 1.0;
        letter-spacing: 0;
      }

      &:focus {
        transition: width .2s ease-in;
        width: 360px;
        font-weight: normal;
        color: $N0;
        border: solid 1px rgba($N0, 0.5) !important;
      }
      &:hover {
        color: $N0;
        border: solid 1px rgba($N0, 0.3);
      }
    }

    :global {
      .qicon {
        --primary-color: #fff;
        //--secondary-color: #d3c4ed;
      }
      .icon {
        color: $N0;
        opacity: 0.5;
        &:hover, &:focus {
          opacity: 1;
          svg {
            --primary-color: #fff;
          }
        }
      }
    }
  }

  .login{
    float: right;
    display: inline-block;
    margin: 14px 0;
    line-height: 20px;
    color: $N500;
    font-weight: 500;
    opacity: 0.9;

    &:hover{
      opacity: 1;
    }
  }

  .user [class*='target']{
    float: right;
    margin: 12px 24px;
    height: 20px;
    width: auto;
    line-height: 20px;
    opacity: 0.9;
    color: $N500;
    font-weight: 500;
    border: 0 none;

    &:hover{
      border: 0 none;
      box-shadow: none;
      opacity: 1;
    }

    :global{
      .icon-caret-down{
        position: relative;
        top: 3px;
        left: 2px;
        svg {
          --primary-color: #{$N500};
          --secondary-color: #{$N500};
        }
      }
    }
  }

  .operateItems{
    min-width: 120px;
    li a{
      display: block;
      height: 32px;
      padding: 0 12px;
      //width: 100%;
      line-height: 32px;
      font-size: $font14;
      font-weight: normal;
      color: $N300;
      border-radius: 0;
      &:hover {
        background-color: $N10;
        color: $N500;
      }
      &.line{
        border-bottom: 1px solid $N10;
      }
    }
  }

  :global{
    .pi-popover-popper{
      margin-left: -62px !important;
      margin-top: 6px !important;
    }
  }
}

.menuOuter {
  .menus{
    float: left;
    margin-left: 50px;
    a{
      display: inline-block;
      margin: 0 12px;
      padding: 13px 0;
      font-size: 14px;
      font-weight: 500;
      line-height: 20px;
      color: $N0;
      &.active{
        border-bottom: 2px solid $Y200;
      }
    }
  }
}
