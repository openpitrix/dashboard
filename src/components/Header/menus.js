export default [
  {
    name: 'App Store',
    link: '',
    introduction: {
      title: 'App Store',
      description: 'MENU_APP_INTRO',
      image: '/menu/store.svg'
    }
  },
  {
    name: 'Purchased',
    link: 'apps',
    introduction: {
      title: 'Purchased',
      description: 'MENU_PURCHASED_INTRO',
      image: '/menu/install.svg'
    }
  },
  {
    name: 'My Instances',
    link: 'clusters',
    introduction: {
      title: 'My Instances',
      description: 'MENU_CLUSTER_INTRO',
      image: '/menu/instance.svg'
    }
  },
  {
    name: 'My Runtimes',
    link: 'runtimes',
    introduction: {
      title: 'My Runtimes',
      description: 'MENU_RUNTIME_INTRO',
      image: '/menu/runtime.png'
    }
  }
];
