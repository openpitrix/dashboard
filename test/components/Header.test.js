import { Provider } from 'mobx-react';
import { BrowserRouter } from 'react-router-dom';
import { observable } from 'mobx';

import Header from 'components/Header';

const renderPage = store => (
  <Provider rootStore={store}>
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  </Provider>
);

describe('Layout/Header', () => {
  let store;

  beforeAll(() => {
    store = observable.object({
      user: {
        isLoggedIn: () => true
      }
    });
  });

  it('basic render: login', () => {
    const wrapper = render(renderPage(store));

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('no login', () => {
    store.user = {
      isLoggedIn: () => false
    };
    const wrapper = render(renderPage(store));

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
