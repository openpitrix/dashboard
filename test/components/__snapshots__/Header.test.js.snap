// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/Header basic render: login 1`] = `
<div
  class="header header menusHeader"
>
  <div
    class="wrapper"
  >
    <a
      href="/"
    >
      <label
        class="logoIcon"
      >
        <img
          class="logo"
          src="/op-logo.svg"
        />
      </label>
    </a>
    <div
      class="menus"
    >
      <a
        aria-current="page"
        class="active active"
        href="/"
      >
        App Store
      </a>
      <a
        class=""
        href="/user/apps"
      >
        Purchased
      </a>
      <a
        class=""
        href="/user/clusters"
      >
        My Instances
      </a>
      <a
        class=""
        href="/user/runtimes"
      >
        My Runtimes
      </a>
    </div>
    <div
      class="user"
    >
      <div
        class="tooltip popover"
      >
        <div
          class="target"
        >
          <span
            class="icon icon-caret-down icon"
            style="width:12px;height:12px"
          >
            <svg
              class="qicon qicon-caret-down qicon-dark"
            >
              <use
                href="#qui-caret-down"
              />
            </svg>
          </span>
        </div>
        <div
          class="popper pi-popover-popper"
          style="position:absolute;pointer-events:none;opacity:0"
        >
          <div>
            <div
              class="pi-popover-content"
            >
              <ul
                class="menuLayer"
              >
                <li>
                  <span
                    class="userIcon"
                  >
                    <span
                      class="icon icon-human iconImg"
                      style="width:20px;height:20px"
                    >
                      <svg
                        class="qicon qicon-human qicon-dark"
                      >
                        <use
                          href="#qui-human"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="username"
                  />
                  <span
                    class="devIconOuter undefined"
                  >
                    <span
                      class="icon icon-enterprise devIcon"
                      style="width:8px;height:8px"
                    >
                      <svg
                        class="qicon qicon-enterprise qicon-white"
                      >
                        <use
                          href="#qui-enterprise"
                        />
                      </svg>
                    </span>
                  </span>
                </li>
                <li
                  class="dev"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-appcenter icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-appcenter qicon-dark"
                      >
                        <use
                          href="#qui-appcenter"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      App Center
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/profile/account"
                  >
                    <span
                      class="icon icon-folder icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-folder qicon-dark"
                      >
                        <use
                          href="#qui-folder"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Account Info
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/profile/password"
                  >
                    <span
                      class="icon icon-lock icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-lock qicon-dark"
                      >
                        <use
                          href="#qui-lock"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Change Password
                    </span>
                  </a>
                </li>
                <li
                  class="disabled"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-loudspeaker icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-loudspeaker qicon-dark"
                      >
                        <use
                          href="#qui-loudspeaker"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Notice settings
                    </span>
                  </a>
                </li>
                <li
                  class="disabled"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-creditcard icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-creditcard qicon-dark"
                      >
                        <use
                          href="#qui-creditcard"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Payment
                    </span>
                  </a>
                </li>
                <li
                  class="divider"
                >
                  <a
                    href="/logout"
                  >
                    <span
                      class="icon icon-previous icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-previous qicon-dark"
                      >
                        <use
                          href="#qui-previous"
                        />
                      </svg>
                    </span>
                    <span>
                      Log out
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout/Header no login 1`] = `
<div
  class="header header menusHeader"
>
  <div
    class="wrapper"
  >
    <a
      href="/"
    >
      <label
        class="logoIcon"
      >
        <img
          class="logo"
          src="/op-logo.svg"
        />
      </label>
      <label
        class="logoName"
      >
        QingCloud App Center
      </label>
    </a>
    <a
      class="login"
      href="/login"
    >
      Sign In
    </a>
    <a
      class="upgrade"
      href="/user/provider/apply"
    >
      <span
        class="icon icon-shield shield"
        style="width:16px;height:16px"
      >
        <svg
          class="qicon qicon-shield qicon-white"
        >
          <use
            href="#qui-shield"
          />
        </svg>
      </span>
      UPGRADE_PROVIDER
    </a>
  </div>
</div>
`;
